First_task/First_task/source/domain/use_cases/PersonLogic.cs

```diff
@@ -0,0 +1,41 @@
+﻿using First_task.source.data;
+using First_task.source.domain.entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+
+namespace First_task.source.domain.use_cases
+{
+    public class PersonLogic
+    {
+        private readonly IPersonRepository _repo;
+
+        public PersonLogic(IPersonRepository repo)
+        {
+            _repo = repo;
+        }
+
+        public void DemonstrateEquals()
+        {
+            IEnumerable<Person> distinctPersons = _repo.Fetch().Distinct();
+            for(int i =0;i <distinctPersons.Count() - 1; i++)
+            {
+                if(distinctPersons.ElementAt(i).Equals(distinctPersons.ElementAt(i+1)))
+                {
+                    throw new Exception("Methods Equals does not work correctly");
+                }
+            }
+        }
+        public string CountPerson()
+        {
+            int countStudents;
+            int countTeachers;
+            countStudents = _repo.Fetch().Where(p => p is Student).Count();
```

Comment:
int countStudents = _repo.Fetch().... not necessary to declare a variable separetly